tests/integration/pilot/multi_version_revision_test.go

```diff
@@ -0,0 +1,144 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilot
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+type installConfig struct {
+	revision string
+	path     string
+}
+
+type revisionedNamespace struct {
+	revision  string
+	namespace namespace.Instance
+}
+
+// TestMultiVersionRevision tests traffic between data planes running under differently versioned revisions
+// should test all possible revisioned namespace pairings to test traffic between all versions
+func TestMultiVersionRevision(t *testing.T) {
+	framework.NewTest(t).
+		RequiresSingleCluster().
+		Features("installation.upgrade").
+		Run(func(ctx framework.TestContext) {
+			testdataDir := filepath.Join(env.IstioSrc, "tests/integration/pilot/testdata/upgrade")
+
+			// keep these at the latest patch version of each minor version
+			// TODO(samnaser) add 1.7.4 once we flag-protection for reading service-api CRDs (https://github.com/istio/istio/issues/29054)
+			installConfigs := []installConfig{
+				{
+					revision: "1-6-11",
+					path:     filepath.Join(testdataDir, "1.6.11-install.yaml"),
+				},
+			}
+
+			// keep track of applied configurations and clean up after the test
+			configs := make(map[string]string)
+			defer func() {
+				for _, config := range configs {
+					ctx.Config().DeleteYAML("istio-system", config)
+				}
+			}()
+
+			nsNoRevision := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "no-revision",
+				Inject: true,
+			})
+			echoNamespaces := []revisionedNamespace{
+				{
+					revision:  "none",
+					namespace: nsNoRevision,
+				},
+			}
+
+			for _, c := range installConfigs {
+				configBytes, err := ioutil.ReadFile(c.path)
+				if err != nil {
+					t.Errorf("failed to read config golden from path %s: %v", c.path, err)
+				}
+				configs[c.revision] = string(configBytes)
+				ctx.Config().ApplyYAMLOrFail(t, i.Settings().SystemNamespace, string(configBytes))
+
+				// create a namespace pointing to each revisioned install
+				ns := namespace.NewOrFail(t, ctx, namespace.Config{
+					Prefix:   fmt.Sprintf("revision-%s", c.revision),
+					Inject:   true,
+					Revision: c.revision,
+				})
+				echoNamespaces = append(echoNamespaces, revisionedNamespace{
+					revision:  c.revision,
+					namespace: ns,
+				})
+			}
+
+			// create an echo instance in each revisioned namespace, all these echo
+			// instances will be injected with proxies from their respective versions
+			builder := echoboot.NewBuilder(ctx)
+			instances := make([]echo.Instance, len(echoNamespaces))
```

Comment:
nit: you can re-use the pre-deployed echo instead of a new "no revision" instance, by using `apps.<some app>`. Just to save 1 more deployment